File: progra3-0583-equipo5-frontend/Frontend_C#/SoftPac/SoftPacModel/TiposDeCambioDTO.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace SoftPac.Model
{
    public class TiposDeCambioDTO
    {
        private int? tipoCambioId;
        private DateTime? fecha;
        private decimal? tasaDeCambio;
        private MonedasDTO monedaOrigen;
        private MonedasDTO monedaDestino;

        public TiposDeCambioDTO()
        {
            this.MonedaOrigen = null;
            this.MonedaDestino = null;
            this.TipoCambioId = null;
            this.Fecha = null;
            this.TasaDeCambio = null;
        }

        public TiposDeCambioDTO(int? tipo_cambio_id, DateTime? fecha, decimal? tasa_de_cambio)
        {
            this.TipoCambioId = tipo_cambio_id;
            this.Fecha = fecha;
            this.TasaDeCambio = tasa_de_cambio;
            this.MonedaOrigen = null;
            this.MonedaDestino = null;
        }
        public TiposDeCambioDTO(int? tipo_cambio_id, DateTime? fecha, decimal? tasa_de_cambio, MonedasDTO moneda_origen, MonedasDTO moneda_destino)
        {
            this.TipoCambioId = tipo_cambio_id;
            this.Fecha = fecha;
            this.TasaDeCambio = tasa_de_cambio;
            this.MonedaOrigen = moneda_origen;
            this.MonedaDestino = moneda_destino;
        }

        public TiposDeCambioDTO(TiposDeCambioDTO other)
        {
            this.TipoCambioId = other.TipoCambioId;
            this.Fecha = other.Fecha;
            this.TasaDeCambio = other.TasaDeCambio;
            this.MonedaOrigen = new MonedasDTO(other.MonedaOrigen);
            this.MonedaDestino = new MonedasDTO(other.MonedaDestino);
        }


        public int? TipoCambioId { get => tipoCambioId; set => tipoCambioId = value; }
        public DateTime? Fecha { get => fecha; set => fecha = value; }
        public decimal? TasaDeCambio { get => tasaDeCambio; set => tasaDeCambio = value; }
        public MonedasDTO MonedaOrigen { get => monedaOrigen; set => monedaOrigen = value; }
        public MonedasDTO MonedaDestino { get => monedaDestino; set => monedaDestino = value; }
    }
}

File: Backend_Java/SoftPacModel/src/main/java/pe/edu/pucp/softpac/model/TiposDeCambioDTO.java
package pe.edu.pucp.softpac.model;

import java.math.BigDecimal;
import java.util.Date;

public class TiposDeCambioDTO {
    private Integer tipo_cambio_id;
    private Date fecha;
    private BigDecimal tasa_de_cambio;
    private MonedasDTO moneda_origen;
    private MonedasDTO moneda_destino;

    public TiposDeCambioDTO(){
        tipo_cambio_id=null;
        fecha=null;
        tasa_de_cambio=null;
        moneda_origen=null;
        moneda_destino=null;
    }

    public TiposDeCambioDTO(Integer tipo_cambio_id, Date fecha, BigDecimal tasa_de_cambio){
        this.tipo_cambio_id=tipo_cambio_id;
        this.fecha=fecha;
        this.tasa_de_cambio=tasa_de_cambio;
        moneda_origen=null;
        moneda_destino=null;
    }
    public TiposDeCambioDTO(Integer tipo_cambio_id, Date fecha, BigDecimal tasa_de_cambio, MonedasDTO moneda_origen, MonedasDTO moneda_destino){
        this.tipo_cambio_id=tipo_cambio_id;
        this.fecha=fecha;
        this.tasa_de_cambio=tasa_de_cambio;
        this.moneda_origen=moneda_origen;
        this.moneda_destino=moneda_destino;
    }
    public TiposDeCambioDTO(TiposDeCambioDTO tipo_de_cambio){
        this.tipo_cambio_id=tipo_de_cambio.tipo_cambio_id;
        this.fecha=tipo_de_cambio.fecha;
        this.tasa_de_cambio=tipo_de_cambio.tasa_de_cambio;
        this.moneda_origen=new MonedasDTO(tipo_de_cambio.moneda_origen);
        this.moneda_destino=new MonedasDTO(tipo_de_cambio.moneda_destino);
    }

    public Integer getTipo_cambio_id() {
        return tipo_cambio_id;
    }

    public void setTipo_cambio_id(Integer tipo_cambio_id) {
        this.tipo_cambio_id = tipo_cambio_id;
    }

    public Date getFecha() {
        return fecha;
    }

    public void setFecha(Date fecha) {
        this.fecha = fecha;
    }

    public BigDecimal getTasa_de_cambio() {
        return tasa_de_cambio;
    }

    public void setTasa_de_cambio(BigDecimal tasa_de_cambio) {
        this.tasa_de_cambio = tasa_de_cambio;
    }

    public MonedasDTO getMoneda_origen() {
        return moneda_origen;
    }

    public void setMoneda_origen(MonedasDTO moneda_origen) {
        this.moneda_origen = moneda_origen;
    }

    public MonedasDTO getMoneda_destino() {
        return moneda_destino;
    }

    public void setMoneda_destino(MonedasDTO moneda_destino) {
        this.moneda_destino = moneda_destino;
    }
}
